src/EnergyPlus/DaylightingManager.hh

```diff
@@ -600,112 +553,62 @@ struct DaylightingManagerData : BaseGlobalStruct
     // int IConstShaded = 0; // The shaded window construction for switchable windows
     Real64 VTDark = 0.0; // Visible transmittance (VT) of electrochromic (EC) windows in fully dark state
                          // Real64 VTMULT = 1.0;  // VT multiplier for EC windows // I don't think this is used anywhere
-
-    Vector3<Real64> WinNorm; // Window outward normal unit vector
-    Vector3<Real64>
-        SunPrime; // Projection of sun vector onto plane (perpendicular to window plane) determined by WinNorm and vector along baseline of window
-    Vector3<Real64> WinNormCrossBase;                         // Cross product of WinNorm and vector along window baseline
-    Dayltg::Illums XEDIRSK;                                   // Illuminance contribution from luminance element, sky-related
-    Dayltg::Illums XAVWLSK;                                   // Luminance of window element, sky-related
-    Vector3<Real64> RAYCOS;                                   // Unit vector from reference point to sun
-    Vector3<Real64> HP;                                       // Hit coordinates, if ray hits
-    Vector3<Real64> DayltgHitObstructionHP;                   // Hit coordinates, if ray hits an obstruction
-    Vector3<Real64> DayltgHitInteriorObstructionHP;           // Hit coordinates, if ray hits an obstruction
-    Vector3<Real64> RN;                                       // Unit vector along ray
-    Vector3<Real64> DayltgHitBetWinObstructionHP;             // Hit coordinates, if ray hits an obstruction surface (m)
-    Vector3<Real64> DayltgHitBetWinObstructionRN;             // Unit vector along ray from R1 to R2
-    Vector2<Real64> DFSUHR;                                   // Sun daylight factor for bare/shaded window
-    Vector2<Real64> BFSUHR;                                   // Sun background luminance factor for bare/shaded window
-    Vector2<Real64> SFSUHR;                                   // Sun source luminance factor for bare/shaded window
+    std::array<Real64, (int)DataSurfaces::WinCover::Num> DFSUHR;    // Sun daylight factor for bare/shaded window
```

Comment:
Use `Vector2` only when you actually mean `{x, y}`. For any other indexed collection of two use `std::array<X, Yenum::Num>` with an enumeration-based index. Or use `std::pair` if `{first, second}` is ok. Limit use of `Vector2` to 2D geometrical information.